### PR TITLE
doc: Add u-boot-env.md

### PR DESCRIPTION
**Pull request by rshym**
This is markdown-ed and slightly reworked edition of
`meta-xt-prod-devel/doc/u-boot-env-txt`.

List of noticable changes:
  - reworked MAC related instructions
  - changed paths to boot files according to moulin-based builds
    (i.e. no '/boot' folder, dom0.dtb -> xen.dtb, xen-Image -> xen)
  - removed tee-raw related paragraph
  - Paragraph 'how to check source of boot of loaders' is merged
    into 'how to select source of boot of loaders'.

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>

### CODE REVIEW

File: doc/u-boot-env.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+If you need to set MAC, use variable `ethaddr` and MAC address provided on sticker on ethernet connector on board.
+Use following format:
+```
+setenv ethaddr 12:34:56:78:9A:BC
+```
+
+## TFTP boot
+```
+setenv bootcmd run bootcmd_tftp
+setenv bootcmd_tftp 'run tftp_xen_load; run tftp_dtb_load; run tftp_kernel_load; run tftp_xenpolicy_load; run tftp_initramfs_load; bootm 0x48080000 0x84000000 0x48000000'
```
Comment (lorc) [inline doc/u-boot-env.md:32]: In this and in the following sections I'd rather change order of variables. Now you are referring to a commands that are not available yet, which makes it hard to parse this in a head.

I think that `bootcmd_tftp` should be introduced after `tftp_xen_load`, `tftp_dtb_load`,.. etc, and `bootcmd` - after `bootcmd_tftp`.

**Review by lorc — COMMENTED** [review]
Generally I'm okay with the documentation, but it would be totally cool to have this env variables set by default in the U-Boot.

If you have time - would you please take a look to `include/configs/rcar-gen3-common.h`, variable `CONFIG_EXTRA_ENV_SETTINGS`?

**Comment by rshym** [comment]
The corresponding patch for u-boot's default environment is https://github.com/xen-troops/meta-xt-rcar/pull/17

File: doc/u-boot-env.md (hunk 1 above)
Comment (rshym) [reply doc/u-boot-env.md]: Thanks for pointing. Fixed according to comment.

**Comment by rshym** [comment]
Documents is reworked, taking into account that all mentioned commands are part of u-boot's default environment.

**Comment by lorc** [comment]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by rshym** [comment]
Reviewer is added. Ready to be merged.

@iusyk could you please merge?
